### PR TITLE
exp/clients/horizon: metrics endpoint

### PR DESCRIPTION
**Pull request by poliha**
This adds support to SDK for getting metrics from the horizon server

### CODE REVIEW

File: exp/clients/horizon/responses.go [hunk 1]
```diff
@@ -179,3 +179,57 @@ type LedgersPage struct {
...
+	RequestsFailed         LogMetric      `json:"requests.failed"`
+	RequestsSucceeded      LogMetric      `json:"requests.succeeded"`
+	RequestsTotal          LogTotalMetric `json:"requests.total"`
+	CoreLatestLedger       SingleMetric   `json:"stellar_core.latest_ledger"`
+	CoreOpenConnections    SingleMetric   `json:"stellar_core.open_connections"`
+	TxsubBuffered          SingleMetric   `json:"txsub.buffered"`
+	TxsubFailed            LogMetric      `json:"txsub.failed"`
+	TxsubOpen              SingleMetric   `json:"txsub.open"`
+	TxsubSucceeded         LogMetric      `json:"txsub.succeeded"`
+	TxsubTotal             LogTotalMetric `json:"txsub.total"`
```
Comment (bartekn) [inline exp/clients/horizon/responses.go]: I think we should move this to `protocols/horizon` and also update Horizon to use this structs.

File: exp/clients/horizon/client.go [hunk 2]
```diff
@@ -104,3 +104,26 @@ func (c *Client) LedgerDetail(sequence uint32) (ledger Ledger, err error) {
...
+	req.Header.Set("X-Client-Version", app.Version())
+
+	resp, err := c.HTTP.Do(req)
+	if err != nil {
+		return
+	}
+
+	err = decodeResponse(resp, &metrics)
+	return
+
```
Comment (bartekn) [inline exp/clients/horizon/client.go]: Can't we use `sendRequest` here? This looks almost the same.

File: exp/clients/horizon/responses.go (hunk 1 above)
Comment (poliha) [reply exp/clients/horizon/responses.go]: Yea I think you are right. I was going to move new response structs to `protocols/horizon` when I am done with the new horizon client, but it seems like its just going to get more complicated then. Will make a new PR to move them and start using `protocols/horizon` directly after this PR.

File: exp/clients/horizon/client.go (hunk 2 above)
Comment (poliha) [reply exp/clients/horizon/client.go]: We can, but will need some modification. Currently `sendRequest` calls `request.buildUrl()` to return an endpoint. If I modify `sendRequest` to  receive the endpoint as a parameter, then it should be fine.  

So `sendRequest(hr HorizonRequest, a interface{})` becomes `sendRequest(endpoint string, a interface{})`

What do you think?

Comment (bartekn) [reply exp/clients/horizon/client.go]: This or you can create a private `metricsRequest` struct that implements `HorizonRequest` interface.

Comment (poliha) [reply exp/clients/horizon/client.go]: Yea that will work as well.. wasn't sure if we wanted to create structs when all we needed was just a string... this has leads to less duplicate code though.. so will go for the structs

File: exp/clients/horizon/responses.go (hunk 1 above)
Comment (poliha) [reply exp/clients/horizon/responses.go]: This will be implemented in #996

**Pull request merged**